application/common/models/User.php

```diff
@@ -0,0 +1,185 @@
+            ],
+            [
+                'email', 'email',
+            ],
+            [
+                'password', 'required',
+                'on' => [self::SCENARIO_UPDATE_PASSWORD, self::SCENARIO_AUTHENTICATE],
+            ],
+            [
+                'password', 'string',
+            ],
+            [
+                'password',
+                $this->validatePassword(),
+                'on' => self::SCENARIO_AUTHENTICATE,
+            ],
+            [
+                ['last_synced_utc', 'last_changed_utc'],
+                'default', 'value' => MySqlDateTime::now(),
+            ],
+        ], parent::rules());
+    }
+
+    private function validatePassword(): \Closure
+    {
+        return function ($attributeName) {
+            if (! password_verify($this->$attributeName, $this->password_hash)) {
+                $this->addError($attributeName, 'Incorrect password.');
+            }
+        };
+    }
+
+    public function behaviors(): array
+    {
+        return [
+            'changeTracker' => [
+                'class' => AttributeBehavior::className(),
+                'attributes' => [
+                    ActiveRecord::EVENT_BEFORE_INSERT => 'last_changed_utc',
+                    ActiveRecord::EVENT_BEFORE_UPDATE => 'last_changed_utc',
+                ],
+                'value' => MySqlDateTime::now(),
+                'skipUpdateOnClean' => true, // only update the column if the model is dirty
+            ],
+            //TODO: further design consideration needed here...this should only update via sync processes.
+            'updateTracker' => [
+                'class' => AttributeBehavior::className(),
+                'attributes' => [
+                    ActiveRecord::EVENT_BEFORE_INSERT => 'last_synced_utc',
+                    ActiveRecord::EVENT_BEFORE_UPDATE => 'last_synced_utc',
+                ],
+                'value' => MySqlDateTime::now(),
+                'skipUpdateOnClean' => false, // always update the column regardless of dirtiness
+            ],
+        ];
+    }
+
+    /**
+     * @return array of fields that should be included in responses.
+     */
+    public function fields(): array
+    {
+        return [
+            'employee_id',
+            'first_name',
+            'last_name',
+            'display_name',
+            'username',
+            'email',
+            'active',
+            'locked',
+        ];
+    }
+
+    public function save($runValidation = true, $attributeNames = null)
+    {
+        if ($this->scenario === self::SCENARIO_UPDATE_PASSWORD) {
+            return $this->savePassword($runValidation, $attributeNames);
+        }
+
+        return parent::save($runValidation, $attributeNames);
+    }
+
+    private function savePassword($runValidation = true, $attributeNames = null): bool
+    {
+        $transaction = ActiveRecord::getDb()->beginTransaction();
+
+        try {
+            if ($this->hasPasswordAlready()) {
```

Comment:
Is this code still just trying to save the previous password to the password history list, or has it been updated to immediately save the new password to the password history list?